solutions/enterprise/variables.tf

```diff
@@ -0,0 +1,254 @@
+  default     = []
+}
+
+##############################################################
+# Context-based restriction (CBR)
+##############################################################
+
+variable "cbr_rules" {
+  type = list(object({
+    description = string
+    account_id  = string
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+  }))
+  description = "A single context-based restriction rule to create. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/enterprise/DA-schemas-topics-cbr.md#options-with-cbr)."
+  default     = []
+  nullable    = false
+  # Additional validation happens in the rule module
+  validation {
+    condition     = !(length(var.cbr_rules) > 1)
+    error_message = "Only one context-based restriction rule is allowed."
+  }
+}
+
+variable "service_credential_names" {
+  description = "The mapping of names and roles for service credentials that you want to create for the Event streams.[Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/enterprise/DA-types.md#svc-credential-name)"
+  type        = map(string)
+  default     = {}
+}
+
+variable "metrics" {
+  type        = list(string)
+  description = "Enhanced metrics to activate, as list of strings. Allowed values: 'topic', 'partition', 'consumers'."
+  default     = []
+
+  validation {
+    condition     = alltrue([for name in var.metrics : contains(["topic", "partition", "consumers"], name)])
+    error_message = "The specified metrics are not valid. The following values are valid for metrics: 'topic', 'partition', 'consumers'."
+  }
+}
+
+variable "quotas" {
+  type = list(object({
+    entity             = string
+    producer_byte_rate = optional(number, -1)
+    consumer_byte_rate = optional(number, -1)
+  }))
+  description = "Quotas to be applied to the Event Streams instance. Entity may be 'default' to apply to all users, or an IAM ServiceID for a specific user. Rates are bytes/second, with -1 meaning no quota. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/enterprise/DA-types.md#quotas)"
+  default     = []
+}
+
+##############################################################
+# Mirroring
+##############################################################
+
+variable "mirroring_topic_patterns" {
+  type        = list(string)
+  description = "The list of the topics to set in instance. Required only if creating mirroring instance."
+  default     = null
+}
+
+variable "mirroring" {
+  description = "Event Streams mirroring configuration. Required only if creating mirroring instance. For more information on mirroring, see https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-mirroring."
```

Comment:
This description needs to point to supporting doc with details on the object schema.

Reply:
done